proto/engine/v1/custom.go

```diff
@@ -0,0 +1,43 @@
+package enginev1
+
+import (
+	"github.com/ethereum/go-ethereum/common/hexutil"
+	ethrpc "github.com/prysmaticlabs/prysm/v3/beacon-chain/rpc/apimiddleware"
+	"strconv"
+)
+
+func NewExecutionPayloadHeaderFromJSON(header *ethrpc.ExecutionPayloadHeaderJson) (*ExecutionPayloadHeader,
+	error) {
+	blockNumber, err := strconv.ParseUint(header.BlockNumber, 10, 64)
+	if err != nil {
+		return nil, err
+	}
+	gasLimit, err := strconv.ParseUint(header.GasLimit, 10, 64)
+	if err != nil {
+		return nil, err
+	}
+	gasUsed, err := strconv.ParseUint(header.GasUsed, 10, 64)
+	if err != nil {
+		return nil, err
+	}
+	timestamp, err := strconv.ParseUint(header.TimeStamp, 10, 64)
+	if err != nil {
+		return nil, err
+	}
+	return &ExecutionPayloadHeader{
+		ParentHash:       hexutil.MustDecode(header.ParentHash),
+		FeeRecipient:     hexutil.MustDecode(header.FeeRecipient),
+		StateRoot:        hexutil.MustDecode(header.StateRoot),
+		ReceiptsRoot:     hexutil.MustDecode(header.ReceiptsRoot),
+		LogsBloom:        hexutil.MustDecode(header.LogsBloom),
+		PrevRandao:       hexutil.MustDecode(header.PrevRandao),
+		BlockNumber:      blockNumber,
+		GasLimit:         gasLimit,
+		GasUsed:          gasUsed,
+		Timestamp:        timestamp,
+		ExtraData:        hexutil.MustDecode(header.ExtraData),
+		BaseFeePerGas:    hexutil.MustDecode(header.BaseFeePerGas),
```

Comment:
Not sure if this works..     
    `BaseFeePerGas`
 is a **uint256** type, which is a "BigInt" (4 uint64 long).   I didn't check `geth`'s hexutil, but it may not handle this correctly.

Reply:
went with this, let me know what you think:

```go
	baseFeePerGas, err := strconv.ParseUint(header.BaseFeePerGas, 10, 64)
	if err != nil {
		return nil, err
	}
	return &v11.ExecutionPayloadHeader{
//...
		BaseFeePerGas:    big.NewInt(0).SetUint64(baseFeePerGas).Bytes(),
```